addon/-private/system/references/has-many.js

```diff
@@ -258,8 +258,8 @@ export default class HasManyReference extends Reference {
   }
 
   _isLoaded() {
-    let hasData = get(this.hasManyRelationship, 'hasData');
-    if (!hasData) {
+    let hasRelationshipDataProperty = get(this.hasManyRelationship, 'hasAnyRelationshipData');
+    if (!hasRelationshipDataProperty) {
```

Comment:
this should probably be noted for a follow up PR to be changed. No tests failed but now that we've clarified what the flags are it is rather obvious that this is not correct.